lib/get-config.js

```diff
@@ -338,6 +338,8 @@ function validateRules(rules, loadedRules, options) {
 
 function validateOverrides(config, options) {
   if (config.overrides) {
+    config.overrides = config.overrides.map((item) => Object.assign({}, item));
```

Comment:
It seems vaguely odd to loop over `config.overides` twice. Maybe we could migrate the `for...of` below to be within this `.map`?

Something like:

```js
    config.overrides = config.overrides.map((overrideConfig) => {
      // If there are keys in the object which are not yet supported by overrides functionality, throw an error.
      let overrideKeys = Object.keys(overrideConfig);
      let containsValidKeys = overrideKeys.every((item) => SUPPORTED_OVERRIDE_KEYS.has(item));
      if (!containsValidKeys) {
        throw new Error(
          'Using `overrides` in `.template-lintrc.js` only supports `files` and `rules` sections. Please update your configuration.'
        );
      }
      // clone a deep copy of the override config to ensure it is not mutated
      let clonedResult = JSON.parse(JSON.stringify(overrideConfig));
	
      // TODO: this should be updated to avoid mutation (mutates for deprecated rules), and the mutation should be moved into `processRules` which is expected to return a new value
      validateRules(clonedResult.rules, config.loadedRules, options);

      clonedResult.rules = processRules(clonedResult);

      return clonedResult;
    });
```

Reply:
Note: I added this TODO in the snippet above, should probably be moved into an issue:

> TODO: this should be updated to avoid mutation (mutates for deprecated rules), and the mutation should be moved into `processRules` which is expected to return a new value